docs/design_decisions.md

```diff
@@ -424,3 +424,24 @@ Dan hoeft de 404 foutcode ook niet gespecificeerd te worden. Als "best practice"
     * 500 Internal Server Error
     * 503 Service Unavailable
     * default
+
+### DD5.24 Gebruik minLength en maxLength voor het afdwingen van de lengte van een waarde
+Indien er voor een waarde een vaste lengte gewenst is en er wordt al gebruik gemaakt van een pattern dan is het logisch om die vaste lengte ook af te dwingen met het pattern.
+We kiezen er echter voor om deze altijd te definiëren met de properties 'minlength' en 'maxLength'.
+
+_**Ratio**_
+
+Wanneer een gebruiker (burger of ambtenaar) iets verkeerd invult kan de applicatie explicieter zijn in de beschrijving van de oorzaak. De yaml code
+
+```yaml
+pattern: "^[ 0-9]*$"
+maxLength: 12
+minLength: 12
+```
+maakt het dus mogelijk om de volgende foutmeldingen te genereren:
+
+* er is een niet-numerieke waarde ingetypt (alleen cijfers zijn toegestaan)
```

Comment:
Kan je niet gewoon als foutmelding geven: `<ingetypte string>` is geen getal van 12 cijfers? Als gebruiker zou ik het prettig vinden om als ik een fout maak het direct goed te kunnen doen ipv typen, submitten en elke keer een foutmelding krijgen dat ik een fout heb gemaakt. De meldingen 'te weinig' en 'teveel ingetypt' zeggen ook niet hoeveel het moeten zijn.

Reply:
Deze Design Decision gaat er niet zozeer om hoe de foutmeldingen er precies uit moeten komen te zien. Het gaat er meer om dat je, doordat je geen gebruik maakt van een door een pattern afgedwongen vaste lengte (bijv. '^[0-9]{12}$') maar gebruik maakt van minLength en maxLength, de mogelijkheid hebt om een duidelijkere foutmelding te genereren. Dus ook de door jou voorgestelde.
Overigens wil ik de hier getoonde voorbeeld foutmeldingen best vervangen door die van jou.

Reply:
Ik denk dat ik de Design Decision dan verkeerd begrijp. Zoals het is nu is beschreven lees ik dat we ervoor kiezen om altijd minLength en maxLength gebruiken omdat we daardoor explicieter zijn in de beschrijving van de oorzaak. En dat er aan de hand van de pattern, minLength en maxLength foutmeldingen kan worden gegenereerd. Dit kan jammer genoeg niet. Dit is (nog steeds) handmatig werk en de reden waarom we de foutmeldingen middels scenarios aan het beschrijven zijn.

Reply:
@MelvLee hoe bedoel je "Dit is (nog steeds) handmatig werk en de reden waarom we de foutmeldingen middels scenarios aan het beschrijven zijn." Foutmeldingen worden door de API geleverd op basis van code, dat kan je "gegenereerd" noemen als je wilt, toch? De essentie is dat een pattern meerdere verschillende soorten validaties in één kan combineren, bijvoorbeeld welke tekens geaccepteerd worden, wat de minimale lengte is en wat de maximale lengte is. Daarop is het heel moeilijk goede gerichte foutmeldingen te bedenken die de gebruiker echt helpen. Zodat je niet aan de eindgebruiker hoeft te vertellen dat de invoer niet voldoet aan patroon '^[0-9]{16}$'.

Reply:
Gegenereerd betekent voor de developer (zowel consumer als de provider) dat hij de (boilerplate) code niet hoeft in te typen. Het is niet zo dat je deze patterns en constraints terug zie in de gegenereerde code. Je moet deze constraints dus zelf toevoegen in de code. Ook is het niet mogelijk om in de OpenAPI spec op te nemen dat je een bepaalde code terug krijgt bij het niet voldoen aan een pattern of constraint. Dit moet apart worden gedocumenteerd en ook door zowel consumer als developer moet er logica worden geïmplementeerd om hiermee om te gaan.
De enige reden (denk ik) dat een consumer developer deze regex patterns wil, is dat hij deze ook kan gebruiken in de consumer applicatie om ervoor te zorgen dat hij niet meer/minder waarden doorlaat.
Ik begrijp heel goed dat het moeilijk is om goede gerichte foutmeldingen te bedenken. Ik denk daarom dat mijn  initiele reactie niet duidelijk is geformuleerd. Ik denk dat het handiger is om in het voorbeeld van de DD voor alle 3 rules dezelfde foutmelding te gebruiken, nl dat de ingetypte string geen getal is van 12 cijfers. Wat ga je namelijk doen als de waarde niet voldoet aan twee van de drie regels? Krijgt je dan beide foutmeldingen? En als je maar één krijgt, wat is dan de volgorde? Zoals het is gedefinieerd in de yaml file?
Ik ben het daarom met je eens dat je niet moet willen dat je altijd een regex pattern moet splitsen.

Reply:
Oke, je bedoelt dus ook dat de foutafhandeling en foutmelding moet worden gespecificeerd per parameter (niet (alleen) per soort fout)

Reply:
@MelvLee, uit je laatste reactie maak ik eigenlijk op dat deze DD niet zo heel veel toegevoegde waarde heeft. 
Foutmeldingen kunnen volgens jou toch niet gegenereerd worden in de code. Daarnaast adviseer je om voor de 3 rules dezelfde foutmelding te gebruiken. Daarom vraag ik me af wat dan nog de toegevoegde waarde is van het gebruik van de 3 rules t.o.v. de pattern "^[ 0-9]{12}$".

Reply:
Voor de developer heeft het zeker geen toegevoegde waarde

Reply:
Wat is nu de slotconclusie ? Maakt het helemaal niet uit of je minlength en maxLength gebruikt dan wel de notatie in het Pattern om dat aan te geven ?